algoliasearch-core/src/test/java/com/algolia/search/JacksonParserTest.java

```diff
@@ -242,6 +242,14 @@ void assertOREDResult(List<List<String>> result) {
     assertThat(result.get(0)).containsSequence("color:yellow");
   }
 
+  void assertOREDListResult(List<List<String>> result) {
```

Comment:
IIUC the filters are `ANDs` instead of `ORs`. Can you confirm @godelized? :)  

```suggestion
  void assertANDEDListResult(List<List<String>> result) {
```